rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport.hpp

```diff
@@ -56,11 +58,15 @@ struct StringHelper<rosidl_typesupport_introspection_c__MessageMembers>
   {
     auto c_string = static_cast<rosidl_generator_c__String *>(data);
     if (!c_string) {
-      fprintf(stderr, "Failed to cast data as rosidl_generator_c__String\n");
+      RCUTILS_LOG_ERROR_NAMED(
+        "rmw_fastrtps_cpp",
+        "Failed to cast data as rosidl_generator_c__String")
```

Comment:
A question unrelated to this PR: The `static_cast` will always succeed. So the error message and the test are a bit bogus. Should it be replaced with a check for `!data` and a corresponding error message instead?